### ISSUE TITLE
Need information on how to add certificate authorities to fastlane

### ISSUE THREAD

**Issue by theothermattm** [fastlane/fastlane#9915]
### New Issue Checklist

- [ X ] Updated fastlane to the latest version
- [ X ] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

When running `fastlane supply init` we are getting an error with SSL connections, and we *know why*. We have SSL inspection running on our corporate network and we need to add a certificate authority to any tools making https/ssl connections. 

The error message below is very helpful, however, we've googled like crazy to find out how to set certificate authorities for fastlane, to know avail.  We need to know the proper place to put certificate authorities so fastlane will use them.

##### Complete output when running fastlane, including the stack trace and command used

```
my-pc:fastlane$ fastlane supply init
fastlane could not check for updates error: SSL_connect returned=1 errno=0 state=error: certificate verify failed
[14:49:11]: -----------------------------------------------------------------------
[14:49:11]: SSL_connect returned=1 errno=0 state=error: certificate verify failed
[14:49:11]: 
[14:49:11]: SSL errors can be caused by various components on your local machine.
[14:49:11]: 
[14:49:11]: The best solution is to use the self-contained fastlane version.
[14:49:11]: Which ships with a bundled OpenSSL,ruby and all gems - so you don't depend on system libraries
[14:49:11]:  - Use Homebrew
[14:49:11]:     - update brew with `brew update`
[14:49:11]:     - install fastlane using:
[14:49:11]:       - `brew cask install fastlane`
[14:49:11]:  - Use One-Click-Installer:
[14:49:11]:     - download fastlane at https://download.fastlane.tools
[14:49:11]:     - extract the archive and double click the `install`
[14:49:11]: -----------------------------------------------------------
[14:49:11]: for more details on ways to install fastlane please refer the documentation:
[14:49:11]: -----------------------------------------------------------
[14:49:11]:         :rocket:       https://docs.fastlane.tools          :rocket:   
[14:49:11]: -----------------------------------------------------------
[14:49:11]: 
[14:49:11]: You can also install a new version of Ruby
[14:49:11]: 
[14:49:11]: - Make sure OpenSSL is installed with Homebrew: `brew update && brew upgrade openssl`
[14:49:11]: - If you use system Ruby:
[14:49:11]:   - Run `brew update && brew install ruby`
[14:49:11]: - If you use rbenv with ruby-build:
[14:49:11]:   - Run `brew update && brew upgrade ruby-build && rbenv install 2.3.1`
[14:49:11]:   - Run `rbenv global 2.3.1` to make it the new global default Ruby version
[14:49:11]: - If you use rvm:
[14:49:11]:   - First run `rvm osx-ssl-certs update all`
[14:49:11]:   - Then run `rvm reinstall ruby-2.3.1 --with-openssl-dir=/usr/local`
[14:49:11]: 
[14:49:11]: If that doesn't fix your issue, please google for the following error message:
[14:49:11]:   'SSL_connect returned=1 errno=0 state=error: certificate verify failed'
[14:49:11]: -----------------------------------------------------------------------

[!] SSL_connect returned=1 errno=0 state=error: certificate verify failed
```

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.11.0 (Apple Git-81)           |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.12.6 (16G29)                    |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 8.3.3                                       |
### System Locale
| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:
<details><summary>`./Fastfile`</summary>
```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command
# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`
# If you want to automatically update fastlane if a new version is available:
# update_fastlane
# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.36.0"
default_platform :android
platform :android do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
  end
  desc "Runs all the tests"
  lane :test do
    gradle(task: "test")
  end
  desc "Submit a new Beta Build to Crashlytics Beta"
  lane :beta do
    gradle(task: "assembleRelease")
    # sh "your_script.sh"
    # You can also use other beta testing services here
  end
  desc "Deploy a new version to the Google Play"
  lane :deploy do
    gradle(task: "assembleRelease")
    supply
  end
  # You can define as many lanes as you want
  after_all do |lane|
    # This block is called, only if the executed lane was successful
    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end
  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end
# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions
# fastlane reports which actions are used. No personal data is recorded. 
# Learn more at https://github.com/fastlane/fastlane#metrics
```
</details>
<details><summary>`./Appfile`</summary>
```ruby
json_key_file "Google Play Android Developer-6c1e72bf21d4.json" # Path to the json secret file - Follow https://github.com/fastlane/supply#setup to get one
package_name "org.mycompany.mobile.arcadia" # e.g. com.krausefx.app
```
</details>
### fastlane gems
| Gem      | Version | Update-Status  |
| -------- | ------- | -------------- |
| fastlane | 2.46.0  | 💥 Check failed |
### Loaded fastlane plugins:
**No plugins Loaded**
<details><summary><b>Loaded gems</b></summary>
| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.1        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.4        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.6        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.0.15       |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.12.0       |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
</details>
*generated on:* **2017-07-31**
</details>

**Comment by ohayon** [comment]
Hey @theothermattm apologies for the issue you are having! Is it possible that you have a version of fastlane installed via RubyGems in addition to the package that is in your `~/.fastlane` directory? 🐙

**Comment by drao05** [comment]
Hi, @ohayon, I'm one of the devs on this project.
There's no fastlane installed via ruby gems (as per `gem list` ).

**Comment by ohayon** [comment]
Hmmm. Interesting. Thanks for the update @drao05 ! Are you both seeing this issue on different machines?

**Comment by drao05** [comment]
We're seeing this issue on one machine, a mac mini. The specific command we need to run and are getting stuck on is `fastlane supply init`, to sync our android app up with the Google Play Store.

**Comment by ohayon** [comment]
How did you you all install fastlane on to that mac mini?

**Comment by drao05** [comment]
with homebrew - `which fastlane` produces `/Users/[user]/.fastlane/bin/fastlane`

**Comment by ohayon** [comment]
👍  Thanks for the info @drao05 !! We will take a look into that soon and get back to you! Apologies for the issues! 🚀 🐙

**Comment by fllink** [comment]
I ran into the same problem when testing fastlane with a proxy locally to simulate a similar corporate network one of our clients has. I ended up doing
`export SSL_CERT_DIR="/usr/local/etc/openssl/certs"`
before calling fastlane, after putting the necessary `certificate.pem` into this folder and running `/usr/local/opt/openssl/bin/c_rehash` (not sure if this is really necessary). This made fastlane accept my custom proxy certificate.

What made me curious and pointed me in that direction is that when I ran the following lane:
```
lane :test do
	puts OpenSSL::X509::DEFAULT_CERT_DIR
end
```
The output was `/Users/crashlytics/buildAgent/work/2484be8b71880c21/bundle-2.28.3/fastlane_lib/dependencies/ssl/certs` which naturally does not exist on my machine.
